packages/client/src/apis/rewards/RewardsQueryAPI.ts

```diff
@@ -1,3 +1,133 @@
-import { BaseVertexAPI } from '../base';
+import { VertexAbis } from '@vertex-protocol/contracts';
+import { toBigInt } from '@vertex-protocol/utils';
+import {
+  ContractFunctionName,
+  encodeAbiParameters,
+  encodePacked,
+  Hex,
+  prepareEncodeFunctionData,
+} from 'viem';
+import { BaseRewardsAPI } from './BaseRewardsAPI';
+import {
+  ClaimLiquidTokensParams,
+  SatelliteTransactionType,
+  VrtxTokenAmountParams,
+} from './types';
 
-export class RewardsQueryAPI extends BaseVertexAPI {}
+export class RewardsQueryAPI extends BaseRewardsAPI {
+  /**
+   * Get the estimated CCIP fee for the `stakeAs` transaction
+   * @param params
+   * @returns
+   */
+  getEstimatedStakeAsCcipFee(params: VrtxTokenAmountParams) {
+    const address = this.getWalletClientAddress();
+
+    if (!address) {
+      throw new Error('No wallet client address found');
```

Comment:
As a follow up, we may want to extract this to a custom error class (there seems to be a few of this around)